[Feature] 게시물 UI 구현

dudtjs1021ej opened the pull request:
### 📕 Issue

x

### 📙 작업 내역

- [x] 게시물 UI 구현
- [x] ProfileCoordinator 추가
- [x] extension String: 세자리수 단위 콤마 삽입
- [x] extension NavigationItem: BackButton커스텀, rightButton 추가
- [x] 변경된 디자인 반영


### 📘 작업 유형

- [x] 신규 기능 추가
- [ ] 버그 수정
- [ ] 리펙토링
- [ ] 문서 업데이트


### 📋 체크리스트

- [x] Merge 하는 브랜치가 올바른가?
- [x] PR과 관련없는 변경사항이 없는가?
- [x] 내 코드에 대한 자기 검토가 되었는가?
- [ ] 변경사항이 효과적이거나 동작이 작동한다는 것을 보증하는 테스트를 추가하였는가?
- [ ] 새로운 테스트와 기존의 테스트가 변경사항에 대해 만족하는가?


### 📝 리뷰 노트

- Coordinator패턴 처음 써봐서 최대한 기존 코드 참고해서 구현해보았습니다. 미흡한 점 피드백 부탁드립니디!
- NavigationItem 확장해 backButton과 rightButton 커스텀할 수 있도록 구현해보았습니다. 

### 🏞️ 스크린샷

<img src="https://user-images.githubusercontent.com/77915491/210139272-7aaf4019-3ded-4c0b-9ba7-a0b800fbbbbf.png" width=40%>

dudtjs1021ej committed:
✨ [feat] ProfileCoordinator 추가

dudtjs1021ej committed:
✨ [feat] 게시글 UI 구현

- Post폴더와 Profile폴더 생성
- 게시글 UI 구현

dudtjs1021ej committed:
🚚 [style] 네비게이션바 버튼 이미지 추가

dudtjs1021ej committed:
✨ [feat] custom navigationItem 구현

- NavigationBar의 backButton 커스텀
- NavigationBar의 rightButton 커스텀할 수 있게 구현
- ProfileViewController NaivgationBar 숨김 처리
- ProfileViewController bottom 숨김 처리

dudtjs1021ej committed:
✨ [feat] 세자리수 단위 insertComma 구현

dudtjs1021ej committed:
💄 [style] 프로필 UI 게시물 개수 추가

dudtjs1021ej committed:
💄 [style] 게시물 UI 변경

leeari95 commented on .gitmessage.text:
```diff
@@ -0,0 +1,30 @@
+# <타입>: <제목>
+
+##### 제목은 최대 50 글자까지만 입력 ############## -> |
+
+
+# 본문은 위에 작성
+######## 본문은 한 줄에 최대 72 글자까지만 입력 ########################### -> |
+
+# 꼬릿말은 아래에 작성: ex) #이슈 번호
+
+# --- COMMIT END ---
+# <타입> 리스트
+#✨ [feat]    : 기능 추가
+#🐛 [fix]     : 버그 수정
+#♻️ [refactor]: 리팩토링 (함수 분리/이름수정 등 실행 결과의 변경 없이 코드 구조를 재조정)
+#🚚 [style]   : 파일 형식/네이밍, 폴더 구조/네이밍 수정하거나 옮기는 작업 비즈니스 로직에 변경 없음
+#💄 [style]   : 스타일 (UI 스타일 변경) 비즈니스 로직에 변경 없음
+#📝 [docs]    : 문서 (README 등 문서 추가/수정/삭제)
+#✅ [test]    : 테스트 (테스트 코드 추가/수정/삭제: 비즈니스 로직에 변경 없음)
+#🔧 [chore]   : 기타 사소한 변경사항 (빌드 스크립트 수정 등)
+#💡 [comment] : 필요한 주석 추가 및 변경
+#🔥 [remove]  : 파일, 폴더 삭제 작업
+# ------------------
+#     제목 첫 글자를 대문자로
+#     제목은 명령문으로
+#     제목 끝에 마침표(.) 금지
+#     제목과 본문을 한 줄 띄워 분리하기
+#     본문은 "어떻게" 보다 "무엇을", "왜"를 설명한다.
+#     본문에 여러줄의 메시지를 작성할 땐 "-"로 구분
+# ------------------
```
[P5] 엇.. 이 파일은 운영체제 루트 디렉토리쪽에 추가하는 것 일탠데.. 일부로 커밋에 넣어두신 걸까용?

dudtjs1021ej replied:
https://github.com/YAPP-Github/21st-iOS-Team-2-iOS/pull/9/commits/3e0634b37d68a17af2a24f3fd7e77009c4814551
왜인지 갑자기 커밋 템플릿이 함께 커밋 됐네요..😨 삭제 했습니다!!

leeari95 commented on Fitfty/Projects/Common/Resources/Images.xcassets/btnArrowleft.imageset/Contents.json:
```diff
@@ -0,0 +1,23 @@
+{
+  "images" : [
+    {
+      "filename" : "btnArrowleft.png",
+      "idiom" : "universal",
+      "scale" : "1x"
+    },
+    {
+      "filename" : "btnArrowleft@2x.png",
+      "idiom" : "universal",
+      "scale" : "2x"
+    },
+    {
+      "filename" : "btnArrowleft@3x.png",
+      "idiom" : "universal",
+      "scale" : "3x"
+    }
+  ],
+  "info" : {
+    "author" : "xcode",
+    "version" : 1
+  }
+}
```
[P4] 아이콘을 `.png`가 아니라 `.svg` 확장자로 하는건 어떻게 생각하실까요?

dudtjs1021ej replied:
https://github.com/YAPP-Github/21st-iOS-Team-2-iOS/pull/9/commits/22b384d0b47984d09533dd39393cfc3256d132f0
메인피드 리뷰부분에서 에단과 얘기 나눴던 것 같았는데 잊어버렸네요😅
 `.svg` 파일, 싱글 스케일로 수정했습니다!

leeari95 commented on Fitfty/Projects/Common/Sources/Extension/UINavigationItem+extension.swift:
```diff
@@ -0,0 +1,47 @@
+        let barButtonItem = UIBarButtonItem(customView: button)
+        barButtonItem.customView?.translatesAutoresizingMaskIntoConstraints = false
+        
+        if let customView = barButtonItem.customView {
+            NSLayoutConstraint.activate([
+                customView.heightAnchor.constraint(equalToConstant: size),
+                customView.widthAnchor.constraint(equalToConstant: size)
+            ])
+        }
```
[P4] 이 customView가 위에서 만든 button을 의미하는 것 같은데, 버튼에 직접 레이아웃 제약조건을 주는 것이 아니라 바인딩 해서 주는 이유가 무엇인지 궁금합니당! 🥹

dudtjs1021ej replied:
iOS11부터 UIBarButtonItem 레이아웃을 잡을 때 barbuttonItem.customView에서 잡아야 하는 것으로 알고 있습니다. ([참고](https://stackoverflow.com/questions/43073738/change-size-of-uibarbuttonitem-image-in-swift-3))
바인딩은 customView가 옵셔널이라 추출을 위해 사용했어요!

leeari95 commented on Fitfty/Projects/Coordinator/Sources/TabCoordinator.swift:
```diff
@@ -126,15 +126,17 @@ final class TabCoordinator: NSObject, Coordinator, TabCoordinatorProtocol {
-            tabBarController.addChild(navigationController)
+            coordinator.navigationController.tabBarItem = tabBarItem
+            tabBarController.addChild(coordinator.navigationController)
+            print(tabBarController)
```
[P2] 이 print 문은 제거해주시면 좋을 것 같아요 😊

dudtjs1021ej replied:
https://github.com/YAPP-Github/21st-iOS-Team-2-iOS/pull/9/commits/1d8eca0126f9b0035026963b6b590c0a9d51e701
삭제 했습니다!!

leeari95 commented on Fitfty/Projects/Profile/Sources/View/Post/PostInfoView.swift:
```diff
@@ -0,0 +1,82 @@
+    }
+    
+    private func setUpConstraintsLayout() {
+        [hitsImageView, hitsLabel, bookmarkImageView, bookmarkLabel].forEach {
+            addArrangedSubview($0)
+            $0.translatesAutoresizingMaskIntoConstraints = false
+        }
```
[P4] 이 부분은 제가 별도로 Common에  extension으로 UIView, UIStackView를 확장하여 만든 기능이 있는데 참고해주시면 좋을 것 같아요 🙏🏻

dudtjs1021ej replied:
https://github.com/YAPP-Github/21st-iOS-Team-2-iOS/pull/9/commits/1d8eca0126f9b0035026963b6b590c0a9d51e701
너무 편하네요..! 아리 감사해요👍  프로필 UI도 같이 수정 완료!!

leeari95 commented on Fitfty/Projects/Profile/Sources/View/Post/PostInfoView.swift:
```diff
@@ -0,0 +1,82 @@
+            addArrangedSubview($0)
+            $0.translatesAutoresizingMaskIntoConstraints = false
+        }
+        hitsLabel.setContentHuggingPriority(.defaultHigh, for: .horizontal)
+        bookmarkLabel.setContentHuggingPriority(.defaultHigh, for: .horizontal)
```
[P4] Label을 초기화해주는 부분에 들어가 있어도 좋지 않을까요!?

dudtjs1021ej replied:
https://github.com/YAPP-Github/21st-iOS-Team-2-iOS/pull/9/commits/1d8eca0126f9b0035026963b6b590c0a9d51e701
넵 그러네요..!! 코드 수정했습니당ㅎ

leeari95 commented on Fitfty/Projects/Profile/Sources/ViewController/PostViewController.swift:
```diff
@@ -0,0 +1,64 @@
+    }
+    
+    private func setUpConstraintLayout() {
+        view.addSubview(postView)
+        postView.translatesAutoresizingMaskIntoConstraints = false
```
[P4] 이것도 Common에 UIView+extension 파일을 참고해주세요 😆

dudtjs1021ej replied:
https://github.com/YAPP-Github/21st-iOS-Team-2-iOS/pull/9/commits/1d8eca0126f9b0035026963b6b590c0a9d51e701
수정 했어요!

leeari95 commented on Fitfty/Projects/Profile/Sources/View/Post/PostView.swift:
```diff
@@ -0,0 +1,78 @@
+    }
+    
+    private func setUpConstraintLayout() {
+        [postImageView, dateLabel, contentLabel, postInfoView].forEach {
+            addSubview($0)
+            $0.translatesAutoresizingMaskIntoConstraints = false
+        }
```
[P4] 이 부분도 Common에 UIStackView+extension 파일을 참고해주세요!

dudtjs1021ej replied:
https://github.com/YAPP-Github/21st-iOS-Team-2-iOS/pull/9/commits/1d8eca0126f9b0035026963b6b590c0a9d51e701
수정 완료!!

leeari95 commented on Fitfty/Projects/Profile/Sources/View/Post/PostView.swift:
```diff
@@ -0,0 +1,78 @@
+            $0.translatesAutoresizingMaskIntoConstraints = false
+        }
+        
+        NSLayoutConstraint.activate([
+            postImageView.topAnchor.constraint(equalTo: topAnchor),
+            postImageView.leftAnchor.constraint(equalTo: leftAnchor),
+            postImageView.rightAnchor.constraint(equalTo: rightAnchor),
+            postImageView.heightAnchor.constraint(lessThanOrEqualTo: heightAnchor, multiplier: 0.7),
+            
+            postInfoView.topAnchor.constraint(equalTo: topAnchor, constant: 16),
+            postInfoView.leftAnchor.constraint(equalTo: leftAnchor, constant: 20),
+            
+            contentLabel.leftAnchor.constraint(equalTo: leftAnchor, constant: 20),
+            contentLabel.rightAnchor.constraint(equalTo: rightAnchor, constant: -20),
+            contentLabel.topAnchor.constraint(equalTo: postImageView.bottomAnchor, constant: 20),
+            
+            dateLabel.leadingAnchor.constraint(equalTo: contentLabel.leadingAnchor),
+            dateLabel.topAnchor.constraint(equalTo: contentLabel.bottomAnchor, constant: 8)
+            
+        ])
```
[P5] 중간에 공백이 없어도 되지 않을까요? 
개인적으로 저는 공백이 없어도 앞에 view의 이름이 존재하니까 구분하는데 크게 문제 없다는 생각이 들어서요 🤔 
(사소한 의견입니다! 😊)

dudtjs1021ej replied:
컴포넌트가 많거나 새로 추가할때 컴포넌트별로 공백을 넣으면 좀 더 가독성이 좋은 것 같아 넣어봤습니다..!!
지금은 적어서 공백이 없어도 괜찮을 것 같네요😁

leeari95 reviewed (changes requested):
썬~ 고생많으셨습니다! 말씀주신 코디네이터 부분은 크게 문제가 없는 것 같아요 👍
코멘트 확인 후 답글 꼭 부탁드립니다~ 감사합니다! 😆

dudtjs1021ej committed:
🚚 [style] 버튼 이미지 파일 .png에서 .svg로 변경

dudtjs1021ej committed:
📝 [docs] 불필요한 커밋 템플릿 삭제

dudtjs1021ej committed:
♻️ [refactor] 레이아웃 관련 수정

- extension UIView, StackView 참고하여 addSubview 수정
- PostInfoView label의 setContentHuggingPriority 초기화 부분에서 설정
- 불필요한 print문 삭제

leeari95 commented:
수정사항과 코멘트 모두 확인했습니다!
제 의견 적극적으로 반영해주셔서 감사해요 😊
고생하셨습니다~!🙇🏻‍♀️
